commandapi-platforms/commandapi-velocity/commandapi-velocity-plugin/src/main/java/dev/jorel/commandapi/config/VelocityConfigurationAdapter.java

```diff
@@ -0,0 +1,156 @@
+package dev.jorel.commandapi.config;
+
+import org.spongepowered.configurate.CommentedConfigurationNode;
+import org.spongepowered.configurate.ConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+import org.spongepowered.configurate.yaml.NodeStyle;
+import org.spongepowered.configurate.yaml.YamlConfigurationLoader;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+public record VelocityConfigurationAdapter(YamlConfigurationLoader loader, CommentedConfigurationNode config, DefaultVelocityConfig defaultVelocityConfig) implements ConfigurationAdapter<ConfigurationNode> {
+
+	public static VelocityConfigurationAdapter createDummyInstance() {
+		return new VelocityConfigurationAdapter(null, null, null);
+	}
+
+	@Override
+	public void setValue(String key, Object value) {
+		try {
+			node(key).set(value);
+		} catch (SerializationException e) {
+			e.printStackTrace(System.err);
+		}
+	}
+
+	@Override
+	public void setComment(String key, String[] comment) {
+		StringBuilder commentBuilder = new StringBuilder();
+		for (int i = 0; i < comment.length; i++) {
+			if (i > 0) {
+				commentBuilder.append("\n");
+			}
+			commentBuilder.append(comment[i]);
+		}
+		node(key).comment(commentBuilder.toString());
```

Comment:
Is this loop necessary? I believe we could use `String#join`:

```suggestion
		node(key).comment(String.join("\n", comment));
```

I think this is clearer for doing the same thing, and it cuts down the number of code paths that need to be tested by us.

Reply:
I think for this it really doesn't matter. Whenever I come across a "problem" like this my mind jumps straight to loops and I forget `String.join()` exists. I mean, `String.join()` would be a bit cleaner but I don't think it matters that much.